OpenSim/Common/ComponentOutput.h

```diff
@@ -164,6 +164,114 @@ class Output : public AbstractOutput {
 //=============================================================================
 };  // END class Output
 
+/// @name Creating Outputs for your Component
+/// Use these macros at the top of your component class declaration,
+/// near where you declare @ref Property properties.
+/// @{
+/** Create an output for a member function of this component.
+ *  The following must be true about componentMemberFunction, the function
+ *  that returns the output:
+ *
+ *     -# It is a member function of your component.
+ *     -# The member function is const.
+ *     -# It takes only one input, which is `const SimTK::State&`
+ *
+ *  Use #OpenSim_DECLARE_OUTPUT_FLEX if these are not true for you.
+ *
+ *  You must also provide the stage on which the output depends.
+ *
+ *  Here's an example for using this macro:
+ *  @code{.cpp}
+ *  class MyComponent : public Component {
+ *  public:
+ *      OpenSim_DECLARE_OUTPUT(force, double, getForce, SimTK::Stage::Dynamics);
+ *      ...
+ *  };
+ *  @endcode
+ * @see Component::constructOutput()
+ * @relates OpenSim::Output
+ */
+#define OpenSim_DECLARE_OUTPUT(oname, T, func, ostage)                      \
+    /** @name Outputs                                                    */ \
+    /** @{                                                               */ \
+    /** Provides the value of func##() and is available at stage ostage. */ \
+    /** This output was generated with the                               */ \
+    /** #OpenSim_DECLARE_OUTPUT macro.                                   */ \
+    OpenSim_DOXYGEN_Q_PROPERTY(T, oname)                                    \
+    /** @}                                                               */ \
+    /** @cond                                                            */ \
+    bool _unused_output_##oname { constructOutput<T>(#oname, &Self::func, ostage) }; \
+    /** @endcond                                                         */
```

Comment:
minor but instead of `_unused_output...` you could rename it to `has_output_...` since it is a bool anyway. Might be less confusing than having a bool member named `_unused` and has a value of `true`.